core/primitives/src/utils.rs

```diff
@@ -593,7 +590,7 @@ mod tests {
         assert_ne!(
             create_nonce_with_nonce(&base, salt),
             create_hash_upgradable(
-                CREATE_HASH_PROTOCOL_VERSION,
+                ProtocolFeature::CreateHash.protocol_version(),
```

Comment:
Modify the `create_hash_upgradable` function to take `ProtocolFeature` instead of passing in the protocol version directly as an argument and extract the protocol version within the function.